app/gui/view/graph-editor/src/builtin/visualization/native/lazy_text_visualization/text_provider.rs

```diff
@@ -0,0 +1,269 @@
+            });
+
+            grid_data_update <- receive_data.map(|data| {
+                LazyGridData::try_from(data.clone()).ok()
+            });
+            grid_data_update <- grid_data_update.map(|data| data.clone()).unwrap();
+            chunks_update <- grid_data_update.map(|grid_data| grid_data.chunks.clone());
+            eval chunks_update([text_cache](chunks) {
+                for (pos, text) in chunks {
+                    text_cache.borrow_mut().add_item(*pos, text.clone());
+                }
+            });
+
+            output.line_count <+ grid_data_update.map(|grid_data| grid_data.line_count);
+            output.longest_line <+ grid_data_update.map(|grid_data| grid_data.longest_line);
+
+        }
+
+        Self { frp, text_cache, register_access }
+    }
+}
+
+impl TextProvider for BackendTextProvider {
+    fn get_slice(&self, line: usize, chunk_index: usize) -> Option<String> {
+        let y = line as i32;
+        let x = chunk_index as i32;
+        let result = self.text_cache.borrow_mut().get_item(Vector2::new(x, y));
+        self.register_access.emit(());
+        result
+    }
+
+    fn frp(&self) -> &Frp {
+        &self.frp
+    }
+}
+
+/// Crate a preprocessor configuration for the lazy text preprocessor.
+fn lazy_text_preprocessor(
+    grid_position: GridPosition,
+    grids_size: GridSize,
+) -> PreprocessorConfiguration {
+    PreprocessorConfiguration::new(
+        "Standard.Visualization.Preprocessor",
+        "lazy_preprocessor",
+        vec![
+            serde_json::to_string(&grid_position).unwrap(),
+            serde_json::to_string(&grids_size).unwrap(),
+            serde_json::to_string(&(CHARS_PER_CELL as u32)).unwrap(),
+        ],
+    )
+}
+
+
+
+// =============================
+// === Lazy Grid Data Update ===
+// =============================
+
+type Chunk = (GridPosition, String);
+
+/// Struct for deserialising the data sent from the engine.
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+struct LazyGridData {
+    chunks:       Vec<Chunk>,
+    line_count:   u32,
+    longest_line: u32,
+}
+
+impl TryFrom<visualization::Data> for LazyGridData {
+    type Error = visualization::DataError;
+
+    fn try_from(data: visualization::Data) -> Result<Self, Self::Error> {
+        if let visualization::Data::Json { content } = data {
+            if let Some(chunks) = content.get("chunks") {
+                let line_count =
+                    content.get("lines").map_or_default(|l| l.as_u64().unwrap_or_default() as u32);
+                let longest_line = content
+                    .get("max_line_length")
+                    .map_or_default(|l| l.as_u64().unwrap_or_default() as u32);
+                let chunks: Vec<(GridPosition, String)> = chunks
+                    .as_str()
+                    .map(serde_json::from_str)
+                    .transpose()
+                    .unwrap_or_default()
+                    .unwrap_or_default();
```

Comment:
In case of improper data, we are getting defaults here, but we do not emit a warning, which can hide errors - am I right?